freeform_json_to_yaml.py

```diff
@@ -0,0 +1,440 @@
+            patient = c['patient']
+            race = c['raceEthnicity']['preset']
+
+            demographics = {
+                'age': randint(25, 35) if patient in ['Helga', 'Bob', 'Lily', 'Military Mike'] else randint(45, 58),
+                'sex': 'M' if patient in ['Gary', 'Military Mike', 'Bob'] else 'F',
+                'race': race.split(' ')[0] if 'Islander' not in race and 'Indian' not in race else race
+            }
+
+            is_military = False
+            us_military = True
+            for gear in c['militaryGear']:
+                if len(c['militaryGear'][gear]) > 0:
+                    is_military = True
+                    break
+            if c['clothing']['bottomColor'] != 'original' or c['clothing']['topColor'] != 'original':
+                us_military = False
+
+            if is_military:
+                demographics['military_disposition'] = 'Allied US' if us_military else 'Allied'
+                sim_env = self.json_data['scene'].split('-')[1].replace('sub', 'submarine')
+                if sim_env != 'submarine':
+                    demographics['role'] = 'Infantry'
+                demographics['mission_importance'] = 'normal'
+                if us_military:
+                    demographics['military_branch'] = 'US Army' if sim_env in ['jungle', 'desert', 'urban'] else 'US Navy'
+            
+            new_c['demographics'] = demographics
+            vitals = c['vitals']
+            new_c['vitals'] = {
+                'avpu': 'ALERT' if MENTAL_STATUS_MAP[vitals['mood']] != 'UNRESPONSIVE' else 'UNRESPONSIVE',
+                'ambulatory': c['triageStatus']['sort'] == 'walker',
+                'mental_status': MENTAL_STATUS_MAP[vitals['mood']],
+                'breathing': BREATHING_MAP[vitals['breath']],
+                'heart_rate': PULSE_MAP[vitals['pulse']],
+                'spo2': 67 if vitals['SpO2'] == 'none' else 85 if vitals['SpO2'] == 'low' else 95
+            }
+            injuries = []
+            written_injuries = {}
+            for i in c['injuries']:
+                if i['type'] == 'Asthmatic':
+                    injury = {
+                        'name': 'Asthmatic',
+                        'location': "internal",
+                        'status': 'hidden'
+                    }
+                    written_injuries[injury['name']] = ['internal']
+                elif i['type'] == 'Ear Bleed':
+                    injury = {
+                        'name': 'Ear Bleed',
+                        'location': "right face" if randint(1, 2) > 1 else "left face",
+                        'status': 'discoverable'
+                    }
+                    written_injuries[injury['name']] = [injury['location']]
+                else:
+                    if i['type'] == 'Forehead Scrape':
+                        injury = {
+                            'name': "Laceration",
+                            'location': "left face",
```

Comment:
Would the location of the face have to be randomized for left or right like done above for ear bleed?

Reply:
No, because the sim only has a left forehead scrape. The ear bleed does not appear in the sim, so we can make it either, but if we want them to match as closely as possible, the forehead scrape is left on the sim. Good catch though